services/gateway/pkg/config/config.go

```diff
@@ -31,20 +31,20 @@ type Config struct {
 
 	FrontendPublicURL string `yaml:"frontend_public_url" env:"OCIS_URL;GATEWAY_FRONTEND_PUBLIC_URL" desc:"The public facing URL of the oCIS frontend." introductionVersion:"pre5.0"`
 
-	UsersEndpoint             string `yaml:"-"`
-	GroupsEndpoint            string `yaml:"-"`
-	PermissionsEndpoint       string `yaml:"-"`
-	SharingEndpoint           string `yaml:"-"`
-	AuthAppEndpoint           string `yaml:"-"`
-	AuthBasicEndpoint         string `yaml:"-"`
-	AuthBearerEndpoint        string `yaml:"-"`
-	AuthMachineEndpoint       string `yaml:"-"`
-	AuthServiceEndpoint       string `yaml:"-"`
-	StoragePublicLinkEndpoint string `yaml:"-"`
-	StorageUsersEndpoint      string `yaml:"-"`
-	StorageSharesEndpoint     string `yaml:"-"`
-	AppRegistryEndpoint       string `yaml:"-"`
-	OCMEndpoint               string `yaml:"-"`
+	UsersEndpoint             string `yaml:"users_endpoint" env:"GATEWAY_USERS_ENDPOINT" desc:"The USERS API endpoint." introductionVersion:"%%NEXT%%"`
```

Comment:
These are a lot of new envvars. Can't we make it more convenient for devops to configure them? I imagine they don't want to maintain another 13 envvars.

If it's only about the protocol, can't we have a global envvar for that and just add hardcoded service names?

If that is not possible, maybe it makes sense to NOT support env configuration? One could still add them to their `yaml` file if they so choose.

Reply:
Our current default uses the service name. and we need to make this configurable to test dns based load balancing that skips the registry lookup. the grpc client will then use the configured dns:///service or kubernetes:///service name to look up all available endpoints. that is why it has to be 13 different endpoint.

If we decide to change the default for eg unix sockets on single node deployments we need to change the helm charts as well. but since we do not change the default from the existing hardcoded service name this is not breaking the chart.

For testing the load balancing changing env vars in kubernetes is actuallly a lot easier than providing custom yaml files.

So I don't see any inconvenience for devops right now. And if we decide to change the default WE - as in engineering not devops - need to update the helm charts anyway.

Reply:
Ok then fine for me. But if it is real envvars they need a proper description please. `The USERS API endpoint.` is not explanatory enough.